### PR TITLE
[tests-only] Adjust unskipped publicLink tests

### PR DESCRIPTION
**Pull request by saw-jan**
part of https://github.com/owncloud/owncloud-sdk/issues/644 and https://github.com/owncloud/owncloud-sdk/issues/780

Adjust pact interaction for unskipped tests in publicLinkTest.js

Fixes https://github.com/owncloud/owncloud-sdk/issues/788

### CODE REVIEW

File: tests/helpers/providerStateHelper.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ const givenGroupExists = (provider, group) => {
  */
 const givenFileExists = (provider, username, password, resource) => {
   return provider
-    .given('file exists', { username, password, fileName: encodeURIComponent(resource) })
+    .given('file exists', { username, password, fileName: encodeURIComponent(resource).split('%2F').join('/') })
```
Comment (dpakach) [inline tests/helpers/providerStateHelper.js:36]: Why is this required here?, if it doesn't likes encoded filename then we can use `decodeURIComponent`

File: tests/providerTest.js [hunk 2]
```diff
@@ -314,6 +314,29 @@ describe('provider testing', () => {
             date: date
           }
         }
+      },
+      'public link share is created': (setup, parameters) => {
+        if (setup) {
+          const { username, userPassword, path, shareType, permissions, password } = parameters
```
Comment (dpakach) [inline tests/providerTest.js:320]: share type for the public link will always be the same you don't have to pass it from parameters

File: tests/helpers/providerStateHelper.js (hunk 1 above)
Comment (saw-jan) [reply tests/helpers/providerStateHelper.js:36]: It needs to encode files but there can be a path like `folder/file.txt`. In that case, it will also encode `/` which we don't want happen.

**Review by dpakach — COMMENTED** [review]
lgtm :+1:

**Review by Talank — COMMENTED** [review]
LGTM

**Review by individual-it — CHANGES_REQUESTED** [review]
for those tests that pass on the provider side on oc10 please set the first attribute of `createProvider` to false that will put them into the `verified` bucket and fail CI if those tests start to fail later
I don't expect any provider tests pass completely on ocis because the capabilities still return the wrong content type

File: tests/helpers/providerStateHelper.js (hunk 1 above)
Comment (individual-it) [inline tests/helpers/providerStateHelper.js:36]: use it in a function like in https://github.com/owncloud/owncloud-sdk/pull/801/files#diff-a16db6b5aac8f41269353f9be2487f36a04fd3e16286cd2c0edf4ce46a489aa5R24

**Review by individual-it — CHANGES_REQUESTED** [review]

**Review by individual-it — APPROVED** [review]
